.github/workflows/nightly.yml

```diff
@@ -1,43 +1,75 @@
 name: nightly build
 env:
-  # thirdparties
   DOCKER_SERVER: tip-tip-wlan-cloud-docker-repo.jfrog.io
   DOCKER_USER_NAME: ${{ secrets.DOCKER_USER_NAME }}
   DOCKER_USER_PASSWORD: ${{ secrets.DOCKER_USER_PASSWORD }}
-  TESTRAIL_USER_PASSWORD: ${{ secrets.TESTRAIL_USER_PASSWORD }}
-  JFROG_USER_PASSWORD: ${{ secrets.JFROG_USER_PASSWORD }}
-  # AWS credentials
+
   AWS_EKS_NAME: tip-wlan-main
   AWS_DEFAULT_OUTPUT: json
   AWS_DEFAULT_REGION: us-east-2
   AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
   AWS_ACCESS_KEY_ID: ${{ secrets.AWS_CLIENT_ID }}
   AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_CLIENT_KEY }}
+
   # Cloud SDK certs
   CACERT: ${{ secrets.CACERT }}
   CAKEY: ${{ secrets.CAKEY }}
+  # https://stackoverflow.com/questions/59977364/github-actions-how-use-strategy-matrix-with-script
+  testbeds: '[
+    {
+      "number": "01",
+      "ip_address": "10.28.3.6",
+      "ap_model": "ECW5410",
+      "cleanup": false
+    },
+    {
+      "number": "03",
+      "ip_address": "10.28.3.10",
+      "ap_model": "ECW5410",
+      "cleanup": true
+    },
+    {
+      "number": "04",
+      "ip_address": "10.28.3.12",
+      "ap_model": "ECW5410",
+      "cleanup": true
+    }
+  ]'
 
 on:
   workflow_dispatch:
+  pull_request:
```

Comment:
Is this just for testing or will it get merged?

Reply:
i think it shall get merged, why?

Reply:
i'm not sure we want to run a different set of test nightly\on pr

Reply:
Was just wondering cause we didn't trigger the nightly tests for PRs beforehand and I don't see a reason why this changed. But I'm completely fine with doing so.

Reply:
With pytest we should be able to call different set of test cases if we want to that may differ from a nightly set. The idea is PR tests are a quick sanity and nightly for longer sanity.